sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/SessionTokenMismatchRetryPolicy.java

```diff
@@ -129,12 +135,18 @@ private boolean shouldRetryLocally(SessionRetryOptions sessionRetryOptions, int
             return true;
         }
 
-        CosmosRegionSwitchHint regionSwitchHint = ImplementationBridgeHelpers
-            .CosmosSessionRetryOptionsHelper
-            .getCosmosSessionRetryOptionsAccessor()
+        CosmosRegionSwitchHint regionSwitchHint = sessionRetryOptionsAccessor
             .getRegionSwitchHint(sessionRetryOptions);
 
-        if (regionSwitchHint == null || regionSwitchHint == CosmosRegionSwitchHint.LOCAL_REGION_PREFERRED) {
+        if (regionSwitchHint != CosmosRegionSwitchHint.REMOTE_REGION_PREFERRED) {
+            return true;
+        }
+
+        // For write operations we need to retry at least MinInRegionRetryTimeForWriteOperations
+        // to allow the region to catch up on replication
+        if (this.isWriteOperation  && !this.waitTimeTimeoutHelper.isElapsed(
```

Comment:
so for writes, we will only retry across region if maxRetryInLocalRegion&elapsed

Reply:
Correct - that is intentional